cfme/services/myservice.py

```diff
@@ -132,12 +211,12 @@ def retire(self):
             assert vm_state in ['off', 'unknown']
 
     def retire_on_date(self, retirement_date):
-        navigate_to(self, 'SetRetirement')
-        fill(retirement_form, {'retirement_date': retirement_date},
-             action=form_buttons.save)
-        navigate_to(self, 'Details')
+        view = navigate_to(self, 'SetRetirement')
+        view.retirement_date.fill(retirement_date)
```

Comment:
Yeah, we talked about this on IRC, this needs porting the old angular handling code into the new widgets.